src/macaron/dependency_analyzer/dependency_resolver.py

```diff
@@ -309,6 +267,121 @@ def tool_valid(tool: str) -> bool:
+                return {}
+
+            if isinstance(dep_analyzer, NoneDependencyAnalyzer):
+                logger.info(
+                    "Dependency analyzer is not available for %s",
+                    build_tool.name,
+                )
+                return {}
+
+            # Start resolving dependencies.
+            logger.info(
+                "Running %s version %s dependency analyzer on %s",
+                dep_analyzer.tool_name,
+                dep_analyzer.tool_version,
+                main_ctx.component.repository.fs_path,
+            )
+
+            log_path = os.path.join(
+                global_config.build_log_path,
+                f"{main_ctx.component.report_file_name}.{dep_analyzer.tool_name}.log",
+            )
+
+            # Clean up existing SBOM files.
+            dep_analyzer.remove_sboms(main_ctx.component.repository.fs_path)
+
+            commands = dep_analyzer.get_cmd()
+            working_dirs: Iterable[Path] = build_tool.get_build_dirs(main_ctx.component.repository.fs_path)
+            for working_dir in working_dirs:
+                # Get the absolute path to use as the working dir in the subprocess.
+                working_dir = Path(main_ctx.component.repository.fs_path).joinpath(working_dir)
+
+                try:
+                    # Suppressing Bandit's B603 report because the repo paths are validated.
+                    analyzer_output = subprocess.run(  # nosec B603
+                        commands,
+                        stdout=subprocess.PIPE,
+                        stderr=subprocess.STDOUT,
+                        check=True,
+                        cwd=str(working_dir),
+                        timeout=defaults.getint("dependency.resolver", "timeout", fallback=1200),
+                    )
+                    with open(log_path, mode="a", encoding="utf-8") as log_file:
+                        log_file.write(analyzer_output.stdout.decode("utf-8"))
+
+                except (subprocess.CalledProcessError, subprocess.TimeoutExpired) as error:
+                    logger.error(error)
+                    with open(log_path, mode="a", encoding="utf-8") as log_file:
+                        log_file.write(error.output.decode("utf-8"))
+                except FileNotFoundError as error:
+                    logger.error(error)
+
+                # We collect the generated SBOM as a best effort, even if the build exits with errors.
+                # TODO: add improvements to help the SBOM build succeed as much as possible.
+                deps_resolved |= dep_analyzer.collect_dependencies(str(working_dir))
+
+            logger.info("Stored dependency resolver log for %s to %s.", dep_analyzer.tool_name, log_path)
+
+        # Use repo finder to find more repositories to analyze.
+        if defaults.getboolean("repofinder", "find_repos"):
+            DependencyAnalyzer._resolve_more_dependencies(deps_resolved)
```

Comment:
The repo finder seems to be called only when an SBOM is not available. This function returns [earlier](https://github.com/oracle/macaron/pull/388/files#diff-c95724433501b4a68d8ffe2e35305bff56f2c26c45571f9250641511a8d48269R292) if an SBOM is available and does not try to resolve more dependencies.